### PR TITLE
OCPBUGS#29657: Mention known issues while moving etcd to a different disk

### PR DESCRIPTION
**Pull request by tmalove**
[OCPBUGS-29657](https://issues.redhat.com/browse/OCPBUGS-29657)
This PR also includes a request in [OCPBUGS-30328](), https://issues.redhat.com/browse/OCPBUGS-30328, which affects the same topic, although with an SELinux focus. Both submitters and some Security folks are a part of the discussion to ensure the result accomplishes the goal of both Jiras.

Version(s):
4.12, 4.13, 4.14, 4.15, 4.16

Link to docs preview: [Moving etcd to a different disk](https://71991--ocpdocs-pr.netlify.app/openshift-enterprise/latest/scalability_and_performance/recommended-performance-scale-practices/recommended-etcd-practices.html#move-etcd-different-disk_recommended-etcd-practices) (Updated: 5/3/24)

QE review:
- [ x] QE has approved this change.

### CODE REVIEW

**Comment by tmalove** [comment]
@kenneth-dsouza what are the updates for the `etcd-mc.yml` file that are needed? An example will be great to add to the doc. Thanks!

**Comment by tmalove** [comment]
@oarribas To suggest recreating the control plane nodes with a faster disk SSD v2 before moving etcd might confuse the user, per the article noted by Pushpendra. Perhaps we can add this information after additional testing. Your thoughts? Thanks.

**Comment by kenneth-dsouza** [comment]
When the master is recreated the boot will fail due to missing device mount. Please hold this PR for now.

We might also need to remove the last step from the document once we update the script to find secondary device. 
https://docs.openshift.com/container-platform/4.12/scalability_and_performance/recommended-performance-scale-practices/recommended-etcd-practices.html#move-etcd-different-disk_recommended-etcd-practices

**Comment by jlebon** [comment]
That documented procedure should be reworked to not require users to replace the MachineConfig at the end. The logic needs to be adapted to be made idempotent so that it works equally well on existing nodes and new nodes. The classic way to do this is via a stamp file. So e.g. your imperative logic would end with `touch /var/lib/my-service-ran` and then adding `ConditionPathExists=!/var/lib/my-service-ran` on the unit in question so that it only runs once.

**Comment by kenneth-dsouza** [comment]
@jlebon I agree the documented procedure should be reworked with warnings/limitations outlined.   I have tested the below script which will help us find secondary device and format it.  The script will need to be changed according to environment, we can share few examples.
~~~
/etc/find-secondary-device 
----
#!/bin/bash
set -uo pipefail

for device in /dev/sd*; do
  /usr/sbin/blkid $device &> /dev/null
  if [ $? == 2  ]; then
     echo "secondary device found $device"
     echo "creating filesystem for etcd mount"
     mkfs.xfs -f $device &> /dev/null
     udevadm settle 
     xfs_admin -L var-lib-etcd $device &> /dev/null
     echo LABEL=/dev/disk/by-label/var-lib-etcd > /etc/secondary-dev.env
     exit
  fi 
done
echo "Couldn't find secondary block device!" >&2
exit 1
----

find-secondary-device.service content: 


          [Unit]
          Description=Find secondary device
          DefaultDependencies=false
          After=systemd-udev-settle.service
          Before=local-fs-pre.target
          ConditionPathExists=!/etc/secondary-dev.env <------ script will only run when this file does not exist

          [Service]
          RemainAfterExit=yes
          ExecStart=/etc/find-secondary-device

          Restart=on-failure
          RestartSec=20s <----------- restart after every 20 seconds if /etc/find-secondary-device script fails.

          [Install]
          WantedBy=multi-user.target
        enabled: true
        name: find-secondary-device.service

Mount file: We are mounting via LABEL and the device timeout has been increased to 120s, you can further tune it according to your need.

      - contents: |
          [Unit]
          Before=local-fs.target

          [Mount]
          What=/dev/disk/by-label/var-lib-etcd
          Where=/var/lib/etcd
          Type=xfs
          TimeoutSec=120s

          [Install]
          RequiredBy=local-fs.target
        enabled: true
        name: var-lib-etcd.mount 
~~~

**Comment by kenneth-dsouza** [comment]
Attaching machine-config which I have tested for Azure OCP cluster and Vmware OCP clusters for /dev/sd* disks [etcd-disk-mc-improved.gz](https://github.com/openshift/openshift-docs/files/14381405/etcd-disk-mc-improved.gz)

**Comment by kenneth-dsouza** [comment]
@jlebon  Thanks for the feedback, I will work on the suggestions.  Currently machineset does not have an option to add secondary disk to the node. A user needs to manually add extra disk via cloud console or vendor console. If there is a delay in adding the disk the system will fall in to maintenance mode, hence for now we have given a timeout of 120s which can be tuned as per need and the service would restart if no secondary disk is provided.

**Comment by tmalove** [comment]
@kenneth-dsouza  let me know when your work on @jlebon suggestions/script is complete and I will remove the last step in the procedure, per your first message. Feel free to send me a Slack message for quick responses...thanks!

File: modules/move-etcd-different-disk.adoc [hunk 1]
```diff
@@ -15,6 +15,11 @@ The Machine Config Operator (MCO) is responsible for mounting a secondary disk f
 This procedure does not move parts of the root file system, such as `/var/`, to another disk or partition on an installed node.
 ====
 
+.Limitations
+
+* When the new disk is attached to the cluster, the etcd database is part of the root mount. It is not part of the secondary disk or the intended disk when the master node is recreated. As a result, the master node will not create a separate `/var/lib/etcd` mount.
+* After the nodes are on a separate disk, you must update the `etcd-mc.yml` file to replace the instances of `_<new_disk_name>_` with the device name.
```
Comment (kowen-rh) [inline modules/move-etcd-different-disk.adoc:21]: This is already mentioned in each applicable step of the procedure. Is mentioning it here redundant, or am I misunderstanding the point of including it here?

**Comment by oarribas** [comment]
> @oarribas To suggest recreating the control plane nodes with a faster disk SSD v2 before moving etcd might confuse the user, per the article noted by Pushpendra. Perhaps we can add this information after additional testing. Your thoughts? Thanks.

@tmalove , in Azure, the control plane nodes can be re-created with 1TB premium SSD Azure disks (P30) in case slower (smaller) ones are in use. Premium SSD are valid as SO disks, and 1TB/1024GB is the default value for the control plane disks in Azure.

For other platforms, the suggestion of recreating the control plane nodes with more performant disks is also valid.


[1] https://learn.microsoft.com/en-us/azure/virtual-machines/disks-types#premium-ssd-size
[2] https://access.redhat.com/solutions/5366631

File: modules/move-etcd-different-disk.adoc (hunk 1 above)
Comment (kenneth-dsouza) [reply modules/move-etcd-different-disk.adoc:21]: We are not mentioning that when a new master node is created via machineset or via iso install the secondary device would not be formatted, as a .mount unit would exist for /var/lib/etcd the system will fail to boot.
Hence we need to know mention limitations, also we are changing the documentation to improve the steps of migrating etcd to secondary device. The current method has lot of issues.

**Comment by kenneth-dsouza** [comment]
> @kenneth-dsouza let me know when your work on @jlebon suggestions/script is complete and I will remove the last step in the procedure, per your first message. Feel free to send me a Slack message for quick responses...thanks!

I will work internally with you on documenting the steps and script.

**Comment by tmalove** [comment]
Preview update: The update link is updated, per a conversation with Kenneth. The remaining change will be to the machine config file in Step 2, which I will get from Kenneth next week and update. Also, the content before the config file will be updated to reference the use of label names versus device names.

**Comment by kenneth-dsouza** [comment]
Attaching the new improved machine-config, incorporated the suggested changes.

[etcd-disk-mc.tar.gz](https://github.com/openshift/openshift-docs/files/15080975/etcd-disk-mc.tar.gz)

**Comment by tmalove** [comment]
Hi @kenneth-dsouza @reivaj84  review the latest preview link with an updated machine config file. Thanks.

**Comment by geliu2016** [comment]
@sandeepknd could you please take a review? thanks

File: modules/move-etcd-different-disk.adoc [hunk 2]
```diff
@@ -10,15 +10,25 @@ You can move etcd from a shared disk to a separate disk to prevent or resolve pe
 
 The Machine Config Operator (MCO) is responsible for mounting a secondary disk for {product-title} {product-version} container storage.
 
+[NOTE]
+====
+This procedure only works with devices where the name starts with `sda` or `vda`.
```
Comment (reivaj84) [inline modules/move-etcd-different-disk.adoc:15]: I don't see this limitation in the base64 code injected by the machine config.
The blkid runs against any sdX device on /dev/

Comment (kenneth-dsouza) [reply modules/move-etcd-different-disk.adoc:15]: In the base64 code we have shown example of sda, but one can change that to /dev/vd*. We can write that as a note and users are free to change it. We can also show the base64 decoded part, if anyone wants to change according to their environment.

**Comment by kenneth-dsouza** [comment]
> @kenneth-dsouza @reivaj84 See the latest update for the device name note and script to change the device name.

Can you change the wording of this sentence "You can change the device name by running the following script:" 
We don't want users to run this script manually,  below is the script used in machine-config to detect secondary device.
In this example the device name is sd*, you can chaneg it to vd* as per need.

In Verification steps we need to change the grep statement to below:

grep /var/lib/etcd /proc/mounts
/dev/sdb /var/lib/etcd xfs rw,seclabel,relatime,attr2,inode64,logbufs=8,logbsize=32k,noquota 0 0

**Comment by kenneth-dsouza** [comment]
For nvme devices we can use  /dev/nvme*[0-9]*n* , this will make sure we only detect block devices. As most aws instances are based on nvme devices. 

~~~
#!/bin/bash
set -uo pipefail

for device in /dev/nvme*[0-9]*n*; do
  /usr/sbin/blkid $device &> /dev/null
  if [ $? == 2  ]; then
     echo "secondary device found $device"
     echo "creating filesystem for etcd mount"
     mkfs.xfs -L var-lib-etcd -f $device &> /dev/null
     udevadm settle 
     touch /etc/var-lib-etcd-mount
     exit 
  fi 
done
echo "Couldn't find secondary block device!" >&2
exit 77
~~~
This way we have tested this machine-config for sd*, vd* and nvme*[0-9]*n*

**Comment by sandeepknd** [comment]
> For nvme devices we can use /dev/nvme*[0-9]_n_ , this will make sure we only detect block devices. As most aws instances are based on nvme devices.
> 
> ```
> #!/bin/bash
> set -uo pipefail
> 
> for device in /dev/nvme*[0-9]*n*; do
>   /usr/sbin/blkid $device &> /dev/null
>   if [ $? == 2  ]; then
>      echo "secondary device found $device"
>      echo "creating filesystem for etcd mount"
>      mkfs.xfs -L var-lib-etcd -f $device &> /dev/null
>      udevadm settle 
>      touch /etc/var-lib-etcd-mount
>      exit 
>   fi 
> done
> echo "Couldn't find secondary block device!" >&2
> exit 77
> ```
> 
> This way we have tested this machine-config for sd*, vd* and nvme*[0-9]_n_

It is successfully executed on aws platform as well.

**Comment by tmalove** [comment]
@kenneth-dsouza @reivaj84 @sandeepknd See the latest preview...thanks.

File: modules/move-etcd-different-disk.adoc [hunk 3]
```diff
@@ -50,146 +82,91 @@ metadata:
...
+            ExecCondition=/usr/bin/test ! -d /var/lib/etcd/member
+            ExecStart=/usr/sbin/setsebool -P rsync_full_access 1
+            ExecStart=/bin/rsync -ar /sysroot/ostree/deploy/rhcos/var/lib/etcd/ /var/lib/etcd/
+            ExecStart=/usr/sbin/semanage fcontext -a -t container_var_lib_t '/var/lib/etcd(/.*)?'
+            ExecStart=/usr/sbin/setsebool -P rsync_full_access 0
+            TimeoutSec=0
+
+            [Install]
+            WantedBy=multi-user.target graphical.target
+        - name: restorecon-var-lib-etcd.service
```
Comment (reivaj84) [inline modules/move-etcd-different-disk.adoc:147]: just 1 thing here, wouldn't this restorecon run every time ? if so, is this necessary? 
other than this, the rest looks good to me.

Comment (tmalove) [reply modules/move-etcd-different-disk.adoc:147]: > just 1 thing here, wouldn't this restorecon run every time ? if so, is this necessary? other than this, the rest looks good to me.

@kenneth-dsouza your thoughts?

Comment (kenneth-dsouza) [reply modules/move-etcd-different-disk.adoc:147]: Yes, but it should not cause any harm. I have kept the same behavior as the previous releases.

**Comment by tjungblu** [comment]
/lgtm 

also from my side

File: modules/move-etcd-different-disk.adoc [hunk 4]
```diff
@@ -50,146 +82,91 @@ metadata:
...
-+
-[source,terminal]
-----
-$ oc replace -f etcd-mc.yml
-----
 
 .Verification steps
 
-* Run the `grep <new_disk_name> /proc/mounts` command in a debug shell for the node to ensure that the disk mounted:
+* Run the `grep /var/lib/etcd /proc/mounts` command in a debug shell for the node to ensure that the disk is mounted:
```
Comment (kowen-rh) [inline modules/move-etcd-different-disk.adoc:169]: Consistency: You replaced "in a debug shell" with "at a debug shell prompt" earlier in the document.

File: modules/move-etcd-different-disk.adoc [hunk 5]
```diff
@@ -10,20 +10,52 @@ You can move etcd from a shared disk to a separate disk to prevent or resolve pe
 
 The Machine Config Operator (MCO) is responsible for mounting a secondary disk for {product-title} {product-version} container storage.
 
+[NOTE]
+====
+This procedure supports NVME devices and devices where the name starts with `/dev/sd*` or `/dev/vd*`. For an NVME device, use `/dev/nvme*[0-9]\*n*` as the device name.
```
Comment (kowen-rh) [inline modules/move-etcd-different-disk.adoc:15]: I'm afraid I don't understand what this admonition means or is trying to say.

Are we saying that the NVME device name must start with `sd*` or `vd*`, or are we suggesting that the reader will have to replace `sd*`/`vd*` with `nvme[0-9]`?

File: modules/move-etcd-different-disk.adoc [hunk 6]
```diff
@@ -10,20 +10,52 @@ You can move etcd from a shared disk to a separate disk to prevent or resolve pe
...
+====
+
+This script is for `/etc/find-secondary-device`. Replace the device name according to your environment. Decode and replace the device name in the machine configuration file.
```
Comment (kowen-rh) [inline modules/move-etcd-different-disk.adoc:18]: Including a script as part of the introductory information feels strange to me. If the script described here is used in the procedure, should creating it be the first step?

**Review by kowen-rh — COMMENTED** [review]
Some comments, but I'm mostly concerned about not understanding the admonition at the beginning of the document.

/remove-label peer-review-in-progress
/label peer-review-done

File: modules/move-etcd-different-disk.adoc (hunk 6 above)
Comment (kenneth-dsouza) [reply modules/move-etcd-different-disk.adoc:18]: We don't need to create the script, it is already part of the machine-config but it is encoded. In order for the users to be aware of the script we have shown it. Also the script needs changes as per customer environment. If the user has disks which starts with /dev/sd* or /dev/vd* or for NVME, they need to change the device name in the script accordingly.

File: modules/move-etcd-different-disk.adoc (hunk 5 above)
Comment (kenneth-dsouza) [reply modules/move-etcd-different-disk.adoc:15]: The end user needs to replace the disk name in the script as per their env.  We have given examples for sd* , vd* and for nvme they can use `/dev/nvme*[0-9]\*n*`

File: modules/move-etcd-different-disk.adoc (hunk 6 above)
Comment (oarribas) [reply modules/move-etcd-different-disk.adoc:18]: If the script could need changes, I think it will be better to explain what needs to be changed and how to encode it before generating the Machine Config to avoid confusion. I understand it should be before the current step 2, right?

File: modules/move-etcd-different-disk.adoc [hunk 7]
```diff
@@ -10,20 +10,60 @@ You can move etcd from a shared disk to a separate disk to prevent or resolve pe
...
+This procedure only supports device names for the following device types:
+
+* SCSI or SATA- `/dev/sd*`
++
+* Virtual - `/dev/vd*`
++
+* NVMe - `/dev/nvme*[0-9]\*n*`
+====
+
+A script for `/etc/find-secondary-device`, similar to the following sample, is encoded in the machine configuration file. Decode and replace the device name in the script according to your environment.
```
Comment (kowen-rh) [inline modules/move-etcd-different-disk.adoc:24]: If I understand this correctly, the script is not encoded into the machine configuration file until that step is done by the reader. I also don't think we need to decode anything unless I'm misunderstanding how machine configuration files work (which is absolutely possible, haha).

I still think that creating the script and base64-encoding it should be the second step of this procedure directly following the `lsblk` verification step.

As I understand it, essentially, the reader needs to take the contents of the script and encode them in base64. That encoded string is then included directly in the `etcd-mc.yml` file. This is the  `source: data:text/plain;charset=utf-8;base64,<lots_of_base64_noise>` string in the YAML. Personally, I think we should leave the actual base64-encoded script out of the docs, as this ensures that the reader _must_ follow the documentation instead of just copying and pasting something that may or may not work in unexpected ways.

**TL;dr:** I'd write something like the following as the second step of the procedure:

```adoc
. Encode the following script in base64, noting the output:
+
[source,bash]
----
#!/bin/bash
set -uo pipefail

for device in <device_type_glob>; do # <1>
[...]
----
<1> Replace `<device_type_glob>` with a shell glob for your block device type. For SCSI or SATA drives, use `/dev/sd*`; for virtual drives, use `/dev/vd*`; for NVMe drives, use `/dev/nvme*[0-9]\*n*`.
```

**Review by kowen-rh — COMMENTED** [review]
I think we could use some changes here to improve the clarity and technical accuracy of the content, personally, but feel free to disagree. I'm not certain that I'm correct about everything I'm noting here, so I would also appreciate it if @kenneth-dsouza or someone in QE could check.

I hope this helps, and note that we probably don't have to make all of these changes immediately! I know we're trying to get this content in ASAP. :slightly_smiling_face:

File: modules/move-etcd-different-disk.adoc (hunk 7 above)
Comment (kenneth-dsouza) [reply modules/move-etcd-different-disk.adoc:24]: Yes, sounds good to me.

File: modules/move-etcd-different-disk.adoc [hunk 8]
```diff
@@ -159,37 +90,91 @@ metadata:
...
   config:
     ignition:
-      version: 3.4.0
+      version: 3.1.0
+    storage:
+      files:
+        - path: /etc/find-secondary-device
+          mode: 0755
+          contents:
+            source: data:text/plain;charset=utf-8;base64,IyEvYmluL2Jhc2gKc2V0IC11byBwaXBlZmFpbAoKZm9yIGRldmljZSBpbiAvZGV2L3NkKjsgZG8KICAvdXNyL3NiaW4vYmxraWQgJGRldmljZSAmPiAvZGV2L251bGwKICBpZiBbICQ/ID09IDIgIF07IHRoZW4KICAgICBlY2hvICJzZWNvbmRhcnkgZGV2aWNlIGZvdW5kICRkZXZpY2UiCiAgICAgZWNobyAiY3JlYXRpbmcgZmlsZXN5c3RlbSBmb3IgZXRjZCBtb3VudCIKICAgICBta2ZzLnhmcyAtTCB2YXItbGliLWV0Y2QgLWYgJGRldmljZSAmPiAvZGV2L251bGwKICAgICB1ZGV2YWRtIHNldHRsZSAKICAgICB0b3VjaCAvZXRjL3Zhci1saWItZXRjZC1tb3VudAogICAgIGV4aXQgCiAgZmkgCmRvbmUKZWNobyAiQ291bGRuJ3QgZmluZCBzZWNvbmRhcnkgYmxvY2sgZGV2aWNlISIgPiYyCmV4aXQgNzcK
```
[the remark this answers is not in the document]
Comment (kenneth-dsouza) [reply modules/move-etcd-different-disk.adoc:99]: Yes, this will help copy-paste issues for users with different environment.

File: modules/move-etcd-different-disk.adoc [hunk 9]
```diff
@@ -37,116 +77,7 @@ $ oc debug node/<node_name>
...
-The nodes are updated and rebooted. After the reboot completes, the following events occur:
-+
-* An XFS file system is created on the specified disk.
-* The disk mounts to `/var/lib/etcd`.
-* The content from `/sysroot/ostree/deploy/rhcos/var/lib/etcd` syncs to `/var/lib/etcd`.
-* A restore of `SELinux` labels is forced for `/var/lib/etcd`.
-* The old content is not removed.
-
-. After the nodes are on a separate disk, update the `etcd-mc.yml` file with contents such as the following, replacing instances of `<new_disk_name>` with the noted device name:
+. Create a `MachineConfig` YAML file named `etcd-mc.yml` as follows. This machine configuration file is for device names that begin with `/dev/sd*`.
```
[the remark this answers is not in the document]
Comment (kowen-rh) [reply modules/move-etcd-different-disk.adoc:80]: I still see the sentence I referred to above. I think this can just be the first sentence and maintain meaning. :slightly_smiling_face: 

```adoc
. Create a `MachineConfig` YAML file named `etcd-mc.yml` with contents such as the following:
```

I added a little bit to it to be consistent with how some other areas of the etcd docs are written.

File: modules/move-etcd-different-disk.adoc [hunk 10]
```diff
@@ -159,37 +90,91 @@ metadata:
...
   config:
     ignition:
-      version: 3.4.0
+      version: 3.1.0
+    storage:
+      files:
+        - path: /etc/find-secondary-device
+          mode: 0755
+          contents:
+            source: data:text/plain;charset=utf-8;base64,<encoded_etc_find_secondary_device_script> # <1>
```
Comment (kowen-rh) [inline modules/move-etcd-different-disk.adoc:99]: Looking at the preview, I can see that the callout actually extends beyond what is normally visible to the reader without scrolling. :disappointed: 

I don't think that it's critical that we do anything about that _right now_, but it's worth knowing about.

File: modules/move-etcd-different-disk.adoc [hunk 11]
```diff
@@ -10,20 +10,60 @@ You can move etcd from a shared disk to a separate disk to prevent or resolve pe
 
 The Machine Config Operator (MCO) is responsible for mounting a secondary disk for {product-title} {product-version} container storage.
 
+[NOTE]
+====
+This procedure only supports device names for the following device types:
```
[the remark this answers is not in the document]
Comment (kowen-rh) [reply modules/move-etcd-different-disk.adoc:15]: If we're keeping the admonition, I'd modify the opening sentence some. Technically, it's the script that we're going to encode that supports certain shell globs for certain device types, not the procedure.

Personally, I would also use definition lists, full sentences, or a table with named headings to list the appropriate globs for device types. This part is a minor concern that we can address later. :slightly_smiling_face:

File: modules/move-etcd-different-disk.adoc [hunk 12]
```diff
@@ -37,116 +77,7 @@ $ oc debug node/<node_name>
 +
 Note the device name of the new disk reported by the `lsblk` command.
```
Comment (kowen-rh) [inline modules/move-etcd-different-disk.adoc:79]: We're missing the step that I suggested in my last review here. :slightly_smiling_face: 

```adoc
. Encode the following script in base64, noting the output:
+
[source,bash]
----
#!/bin/bash
set -uo pipefail

for device in <device_type_glob>; do # <1>
----
<1> Replace `<device_type_glob>` with a shell glob for your block device type. For SCSI or SATA drives, use `/dev/sd*`; for virtual drives, use `/dev/vd*`; for NVMe drives, use `/dev/nvme*[0-9]\*n*`.
```

Note that I've cut the contents of the script short just for readability in a comment. :slightly_smiling_face:

File: modules/move-etcd-different-disk.adoc [hunk 13]
```diff
@@ -10,20 +10,60 @@ You can move etcd from a shared disk to a separate disk to prevent or resolve pe
...
+This procedure only supports device names for the following device types:
+
+* SCSI or SATA- `/dev/sd*`
++
+* Virtual - `/dev/vd*`
++
+* NVMe - `/dev/nvme*[0-9]\*n*`
+====
+
+This script is for `/etc/find-secondary-device`. Decode and replace the device name in the script according to your environment.
```
Comment (kowen-rh) [inline modules/move-etcd-different-disk.adoc:24]: I don't think we need these sentences or the script in the introductory content since encoding the script is a step of the procedure. :slightly_smiling_face:

File: modules/move-etcd-different-disk.adoc [hunk 14]
```diff
@@ -159,37 +90,91 @@ metadata:
...
+            [Service]
+            Type=oneshot
+            RemainAfterExit=yes
+            ExecStart=/sbin/restorecon -R /var/lib/etcd/
+            TimeoutSec=0
+
+            [Install]
+            WantedBy=multi-user.target graphical.target
+----
+<1> Use the encoded string that you previously created and replace it for `<encoded_etc_find_secondary_device_script>`.
```
Comment (kowen-rh) [inline modules/move-etcd-different-disk.adoc:173]:
```adoc
<1> Replace `<encoded_etc_script>` with the encoded script output that you noted.
```

File: modules/move-etcd-different-disk.adoc (hunk 10 above)
Comment (kowen-rh) [reply modules/move-etcd-different-disk.adoc:99]: Oh, we can shorten the name to `<encoded_etcd_script>` or similar! I did this later in the review (line 173), but forgot to update this comment.

**Review by kowen-rh — APPROVED** [review]
LGTM! :+1:

**Review by skopacz1 — APPROVED** [review]
Merge review LGTM, nicely done!
